src/core/mac/mac.cpp

```diff
@@ -2364,6 +2365,33 @@ void Mac::ProcessCsl(const RxFrame &aFrame, const Address &aSrcAddr)
 }
 #endif // OPENTHREAD_CONFIG_MAC_CSL_TRANSMITTER_ENABLE
 
+#if OPENTHREAD_CONFIG_MLE_LINK_METRICS_ENABLE
+void Mac::ProcessEnhAckProbing(const RxFrame &aFrame, const Neighbor &aNeighbor)
+{
+    enum
+    {
+        kEnhAckProbingIeMaxLen = 2,
+    };
+
+    const HeaderIe *enhAckProbingIe =
+        reinterpret_cast<const HeaderIe *>(aFrame.GetThreadIe(ThreadIe::kEnhAckProbingIe));
+    uint8_t data[kEnhAckProbingIeMaxLen];
+    uint8_t dataLen = 0;
+
+    VerifyOrExit(enhAckProbingIe != nullptr);
+
+    dataLen = enhAckProbingIe->GetLength() - sizeof(VendorIeHeader);
+    VerifyOrExit(dataLen <= kEnhAckProbingIeMaxLen);
+
+    memcpy(data, reinterpret_cast<const uint8_t *>(enhAckProbingIe) + sizeof(HeaderIe) + sizeof(VendorIeHeader),
+           dataLen);
+
+    Get<LinkMetrics>().ProcessEnhAckIeData(data, dataLen, aNeighbor);
```

Comment:
I think we are copying the IE data from the `RxFrame` into `data[]` array and then pass it to `ProcessEnhAckIeData()`. And the `ProcessEnhAckIeData()` is treating it as `const uint8_t *` (so cannt change it).

A suggestion for an optimization here: 
Can we instead of copying it, find a pointer to to start of data (in the `RxFrame` buffer) and pass the pointer to `ProcessEnhAckIeData()`.